internal/provider/resource_private_key_test.go

```diff
@@ -145,181 +63,31 @@ func TestPrivateKeyECDSA(t *testing.T) {
-					}
 				`,
-				Check: func(s *terraform.State) error {
-					// Check `.private_key_pem`
-					gotPrivateUntyped := s.RootModule().Outputs["private_key_pem"].Value
-					gotPrivate, ok := gotPrivateUntyped.(string)
-					if !ok {
-						return fmt.Errorf("output for \"private_key_pem\" is not a string")
-					}
-					if !strings.HasPrefix(gotPrivate, "-----BEGIN EC PRIVATE KEY----") {
-						return fmt.Errorf("private key is missing EC key PEM preamble")
-					}
-
-					// Check `.public_key_pem`
-					gotPublicUntyped := s.RootModule().Outputs["public_key_pem"].Value
-					gotPublic, ok := gotPublicUntyped.(string)
-					if !ok {
-						return fmt.Errorf("output for \"public_key_pem\" is not a string")
-					}
-					if !strings.HasPrefix(gotPublic, "-----BEGIN PUBLIC KEY----") {
-						return fmt.Errorf("public key is missing public key PEM preamble")
-					}
-
-					// Check `.public_key_openssh`
-					gotPublicSSHUntyped := s.RootModule().Outputs["public_key_openssh"].Value
-					gotPublicSSH, ok := gotPublicSSHUntyped.(string)
-					if !ok {
-						return fmt.Errorf("output for \"public_key_openssh\" is not a string")
-					}
-					if !strings.HasPrefix(gotPublicSSH, "ecdsa-sha2-nistp256 ") {
-						return fmt.Errorf("SSH public key is missing ecdsa-sha2-nistp256 prefix")
-					}
-
-					// Check `.private_key_openssh`
-					gotPrivateSSHUntyped := s.RootModule().Outputs["private_key_openssh"].Value
-					gotPrivateSSH, ok := gotPrivateSSHUntyped.(string)
-					if !ok {
-						return fmt.Errorf("output for \"private_key_openssh\" is not a string")
-					}
-					if !strings.HasPrefix(gotPrivateSSH, "-----BEGIN OPENSSH PRIVATE KEY----") {
-						return fmt.Errorf("private key is missing RSA key OPENSSH PEM preamble")
-					}
-
-					// Check `.public_key_fingerprint_md5`
-					gotPublicFingerprintMD5Untyped := s.RootModule().Outputs["public_key_fingerprint_md5"].Value
-					gotPublicFingerprintMD5, ok := gotPublicFingerprintMD5Untyped.(string)
-					if !ok {
-						return fmt.Errorf("output for \"public_key_fingerprint_md5\" is not a string")
-					}
-					if gotPublicFingerprintMD5[2] != ':' {
-						return fmt.Errorf("MD5 public key fingerprint is missing ':' in the correct place")
-					}
-
-					// Check `.public_key_fingerprint_sha256`
-					gotPublicFingerprintSHA256Untyped := s.RootModule().Outputs["public_key_fingerprint_sha256"].Value
-					gotPublicFingerprintSHA256, ok := gotPublicFingerprintSHA256Untyped.(string)
-					if !ok {
-						return fmt.Errorf("output for \"public_key_fingerprint_sha256\" is not a string")
-					}
-					if !(strings.HasPrefix(gotPublicFingerprintSHA256, "SHA256:")) {
-						return fmt.Errorf("SHA256 public key fingerprint is is missing the expected preamble")
-					}
-
-					return nil
-				},
+				Check: r.ComposeAggregateTestCheckFunc(
+					r.TestMatchResourceAttr("tls_private_key.test", "private_key_pem", regexp.MustCompile(`^-----BEGIN EC PRIVATE KEY----(.|\s)+-----END EC PRIVATE KEY-----`)),
```

Comment:
Nit: Given #197, should these regular expressions explicitly check the end of the strings as well for newlines, etc. to prevent regressions?